### PR TITLE
Document installing Eventing Sugar Controller

### PR DESCRIPTION
**Pull request by n3wscott**
## Proposed Changes 

- Document the install of the Eventing Sugar Controller
- Update the annotation for `Knative Eventing Injection` for namespaces in the mainline get started docs.

### CODE REVIEW

File: docs/install/any-kubernetes-cluster.md [hunk 1]
```diff
@@ -634,15 +634,28 @@ At this point, you have a basic installation of Knative Eventing!
...
+
+   ```bash
+   kubectl apply --filename {{< artifact repo="eventing" file="eventing-sugar-controller.yaml" >}}
+   ```
+
+The Knative Eventing Sugar Controller will react to special labels and annotations and produce Eventing resources. For example:
+
+ - When a Namespace is labeled with `eventing.knative.dev/injection=enabled`, the controller will create a default broker
+   in that namespace.
+ - When a Trigger is annotated with `eventing.knative.dev/injection=enabled`, the controller will create a Broker named
```
Comment (mattmoor) [inline docs/install/any-kubernetes-cluster.md:652]: Remove trailing whitespace:
```suggestion
 - When a Trigger is annotated with `eventing.knative.dev/injection=enabled`, the controller will create a Broker named
```

File: docs/install/any-kubernetes-cluster.md [hunk 2]
```diff
@@ -634,15 +634,28 @@ At this point, you have a basic installation of Knative Eventing!
...
+   ```bash
+   kubectl apply --filename {{< artifact repo="eventing" file="eventing-sugar-controller.yaml" >}}
+   ```
+
+The Knative Eventing Sugar Controller will react to special labels and annotations and produce Eventing resources. For example:
+
+ - When a Namespace is labeled with `eventing.knative.dev/injection=enabled`, the controller will create a default broker
+   in that namespace.
+ - When a Trigger is annotated with `eventing.knative.dev/injection=enabled`, the controller will create a Broker named 
+   by that Trigger in the Trigger's Namespace.
```
Comment (mattmoor) [inline docs/install/any-kubernetes-cluster.md:653]: Remove trailing whitespace:
```suggestion
   by that Trigger in the Trigger's Namespace.
```

File: docs/install/any-kubernetes-cluster.md [hunk 3]
```diff
@@ -634,15 +634,28 @@ At this point, you have a basic installation of Knative Eventing!
...
    <!-- This indentation is important for things to render properly. -->
    {{< tabs name="eventing_extensions" >}}
-{{% tab name="Enable Broker" %}}
+{{% tab name="Sugar Controller" %}}
 
 <!-- Unclear when this feature came in -->
-{{< feature-state version="v0.5" state="alpha" >}}
+{{< feature-state version="v0.16" state="alpha" >}}
+
+The following command installs the Github Source:
```
Comment (vaikas) [inline docs/install/any-kubernetes-cluster.md:642]: sugar controller?

File: docs/eventing/getting-started.md [hunk 4]
```diff
@@ -19,299 +21,377 @@ Use this guide to learn how to create, send, and verify events in Knative. The s
...
+
+   ```sh
+   NAME                   READY   REASON   BROKER    SUBSCRIBER_URI                                                                 AGE
+   goodbye-display        True             default   http://goodbye-display.event-example.svc.cluster.local/                        9s
+   hello-display          True             default   http://hello-display.event-example.svc.cluster.local/                          16s
+   ```
+
+   If the triggers are correctly configured, they will be ready and pointing to
+   the correct **Broker** (the default broker) and **SUBSCRIBER_URI**
+   (triggerName.namespaceName.svc.cluster.local). If this is not the case, see
```
Comment (vaikas) [inline docs/eventing/getting-started.md:373]: might want to add a note that svc.cluster.local might differ depending on the actual cluster installation?

File: docs/eventing/getting-started.md [hunk 5]
```diff
@@ -19,299 +21,377 @@ Use this guide to learn how to create, send, and verify events in Knative. The s
 
 To complete this guide, you will need the following installed and running:
 
-- A [Kubernetes cluster](https://kubernetes.io/docs/concepts/cluster-administration/cluster-administration-overview/) running v1.15 or higher.
+- A
```
Comment (vaikas) [inline docs/eventing/getting-started.md:24]: I'd remove this whole installation bit here and just point to any-kubernetes-cluster.md. They have drifted in the past.

File: docs/eventing/getting-started.md [hunk 6]
```diff
@@ -19,299 +21,377 @@ Use this guide to learn how to create, send, and verify events in Knative. The s
...
-1. Make sure that you have a functioning Kubernetes cluster. See the [Comprehensive Install guide](../install) for more information.
+1. Make sure that you have a functioning Kubernetes cluster. See the
+   [Comprehensive Install guide](../install) for more information.
 
-   - Old versions of Knative Serving doesn't necessarily work well with latest Knative Eventing, so try to install the latest version of Knative Serving.
+   - Old versions of Knative Serving doesn't necessarily work well with latest
+     Knative Eventing, so try to install the latest version of Knative Serving.
 
-   - If your Kubernetes cluster comes with pre-installed Istio, make sure it has `cluster-local-gateway` [deployed](https://github.com/knative/serving/blob/master/DEVELOPMENT.md#deploy-istio). Depending on which Istio version you have, you'd need to apply the `istio-knative-extras.yaml` in the corresponding version folder at [here](https://github.com/knative/serving/tree/{{< branch >}}/third_party).
+   - If your Kubernetes cluster comes with pre-installed Istio, make sure it has
```
Comment (vaikas) [inline docs/eventing/getting-started.md:60]: would you mind adding this ti 'any-kubernetes-cluster.md' section that talks about eventing install.

**Comment by vaikas** [comment]
I'll clean up the install section in a follow on PR.
/lgtm
/approve

**Pull request merged**